extensions/amp-animation/0.1/amp-animation.js

```diff
@@ -291,9 +291,12 @@ export class AmpAnimation extends AMP.BaseElement {
     // The animation will be triggered (in paused state) and seek will happen
     // regardless of visibility
     this.triggered_ = true;
-    this.hasPositionObserver_ = !!invocation.caller &&
-      invocation.caller.tagName === 'AMP-POSITION-OBSERVER';
-    return this.createRunnerIfNeeded_().then(() => {
+    this.positionObserver_ = invocation.caller;
+    let positionObserverArgs = null;
+    if (this.positionObserver_) {
+      positionObserverArgs = this.parseAttributes_(this.positionObserver_);
```

Comment:
By the time position-observer send the seekTo event, it should have already resolved and calculated all these values, you should be able to send computed values as part of the event and access it here. That way `parseAttributes_`, `applyMargins_` and `validateAndResolveMargin_` can be removed.

amp-animation should not have knowledge of amp-position-observer attributes. I think all the coupling should be through the event channel.

Reply:
Done as per offline conversation added a datastructure on Event in `amp-position-observer` which I then access in `amp-animation`